### PR TITLE
customizable prediction

### PR DESCRIPTION
**Pull request by fxliang**
feat:
customizable prediction with 3 para

predictor/db: {string}
predictor/max_candidates: {int}
predictor/max_iteration: {int}

### CODE REVIEW

File: src/predictor.h [hunk 1]
```diff
@@ -23,6 +23,10 @@ class Predictor : public Processor {
  private:
   enum Action { kUnspecified, kSelect, kDelete };
   Action last_action_ = kUnspecified;
+  int page_size_;          // page_size from schema
```
Comment (eagleoflqj) [inline src/predictor.h]: No need to read `page_size_` from schema. Just let user set a value equals to schema's page size.

File: src/predictor.cc [hunk 2]
```diff
@@ -93,10 +130,35 @@ Predictor* PredictorComponent::Create(const Ticket& ticket) {
...
+      auto* config = schema->config();
+      string customized_db;
+      if (config->GetString("predictor/db", &customized_db) &&
+          !customized_db.empty()) {
+        auto db = std::make_unique<PredictDb>(
+            res->ResolvePath(customized_db).string());
+        if (db && db->Load()) {
+          db_ = std::move(db);
+        } else {
+          LOG(INFO) << "failed to load db file: "
```
Comment (eagleoflqj) [inline src/predictor.cc]: This is an ERROR

File: README.md [hunk 3]
```diff
@@ -3,5 +3,14 @@ librime plugin. predict next word.
...
 ## Usage
 * Put `predict.db` in rime user directory.
-* In `*.schema.yaml`, add `predictor` to the list of `engine/processors` before `key_binder`.
+* In `*.schema.yaml`, add `predictor` to the list of `engine/processors` before `key_binder`. or patch the schema with: `engine/processors/@before 0: predictor`
+* 3 parameters for customizing your predictor:
+```yaml
+engine/predictor/db: predict.db         # predict db file in user directory/shared directory
+                                        # if no valid file found, fallback to 'predict.db'
+engine/predictor/max_candidates: 5      # max prediction candidates every time
+                                        # if no valid value set, fallback to page_size of the schema
```
Comment (eagleoflqj) [inline README.md]: If no valid value set, it shouldn't respect page_size. It should show all.

File: src/predictor.cc [hunk 4]
```diff
@@ -9,11 +9,29 @@
...
 
 Predictor::Predictor(const Ticket& ticket, PredictDb* db)
-    : Processor(ticket), db_(db) {
+    : Processor(ticket),
+      db_(db),
+      max_iteration_(0),
+      max_candidates_(0),
+      page_size_(5) {
+  // load max_iteration_ and max_candidates_
+  auto* schema = ticket.schema;
```
Comment (eagleoflqj) [inline src/predictor.cc]: It's better to move this logic to `Create` so there's no need to check schema and get config twice.

**Review by eagleoflqj — CHANGES_REQUESTED** [review]

File: src/predictor.cc (hunk 4 above)
Comment (fxliang) [reply src/predictor.cc]: done

**Pull request merged**